prelude/align.md

```diff
@@ -23,5 +23,26 @@ impl Align {
     pub fn bytes(self) -> BigInt {
         self.raw
     }
+
+    /// Computes the best alignment possible for the given offset
+    /// (the largest power of two that the offset is a multiple of).
+    ///
+    /// N.B., for an offset of `0`, this returns `2^64`.
+    pub fn max_for_offset(offset: Size) -> Align {
+        let exp = match offset.bytes().trailing_zeros() {
+            Some(trailing) => trailing,
+            /// if offset is zero, trailing_zeros returns None.
+            /// in this case we choose the alignment 2^64.
+            None => BigInt::from(64),
+        };
+
+        Align::from_bytes(BigInt::one() << exp)
+    }
+
+    /// Lower the alignment, if necessary, such that the given offset
+    /// is aligned to it (the offset is a multiple of the alignment).
+    pub fn restrict_for_offset(self, offset: Size) -> Align {
+        self.min(Align::max_for_offset(offset))
```

Comment:
Where is `min` coming from?

Reply:
Align impls `Ord`, I've added a comment for it.